src/evict/evict_page.c

```diff
@@ -213,6 +213,21 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t previous_state, uint3
 
     if (0) {
 err:
+        /*
+         * Add back the eviction failed page to the urgent queue, if the page is from history store.
+         * It is possible that checkpoint adds an obsolete page of history store to get it evicted
+         * but because of concurrent access of the page by checkpoint and eviction, it may fail. Add
+         * them back to the queue and try to evict again.
+         *
+         * It is possible that an eviction failure occurs to a page that is not added by a
+         * checkpoint thread, but we consider those scenarios are limited.
+         */
+        if (WT_IS_LAS(S2BT(session))) {
+            WT_IGNORE_RET_BOOL(__wt_page_evict_urgent(session, ref));
+            WT_STAT_CONN_INCR(session, cache_eviction_history_store_pages_queued_urgent_again);
+            __wt_yield();
+        }
```

Comment:
We can't do this in all cases, I suspect the only case where this is appropriate is when `ret == EBUSY`, and probably only if `EBUSY` is returned from `__evict_exclusive()`, but I'm not sure of that.

Further, why are we doing this to every LAS page that fails to evict? Why not only to ones that were on the urgent queue?

Reply:
Yes, I will fix it to use`EBUSY` cases. During the test of `test_gc03` after adding obsolete pages to the eviction, the pages didn't get evicted from either the hazard pointer availability in `__evict_exclusive()` and `__split_internal_lock()` function.

I don't find a way to know whether the page is from the urgent queue or not? And also I thought the failure from LRU pages is rare (maybe I am wrong).